### PR TITLE
Support device specifiers in PTB example

### PR DESCRIPTION
**Pull request by hvy**
Part of https://github.com/chainer/chainer/issues/6661.

Needs
- `unchain_backward` with ChainerX for TBPTT: Probably needs some discussion.
- ~`int / chainerx.ndarray` https://github.com/chainer/chainer/pull/7075 (Edit: merged)~

### CODE REVIEW

**Comment by hvy** [comment]
Note that the examples are broken for unrelated reasons as mentioned here https://github.com/chainer/chainer/issues/6196.

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Please fix `ptb.rst`.
```
20:29:57 /work/chainer/docs/source/examples/ptb.rst:273:end-before pattern not found: args.gpu
```

**Comment by hvy** [comment]
Fixed RST. Also includes some unrelated changes for trailing whitespace/CR.

File: examples/ptb/train_ptb.py [hunk 1]
```diff
@@ -4,9 +4,13 @@
 This code is ported from the following implementation written in Torch.
 https://github.com/tomsercu/lstm
 
+Lines in this file are directly referenced by the RST file
+`docs/source/examples/ptb.rst`. The RST file must be updated in conjunction.
```
Comment (niboshi) [inline examples/ptb/train_ptb.py]: When it should be updated is not stated.
It's better to align with the statement in `glance.py`.

Comment (hvy) [reply examples/ptb/train_ptb.py]: Thanks for the tip. Fixed as suggested.

**Comment by hvy** [comment]
PTAL.

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
LGTM

**Pull request merged**